### PR TITLE
Editor & object initialization

### PR DESCRIPTION
**Pull request by AltenGD**

### CODE REVIEW

**Comment by AlFasGD** [comment]
You are really trying to avenge me with 1.6k files to review, I see

File: GDEdit/GDE.App/Main/UI/Toolbar/ToolbarMenu.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            }
+
+            protected sealed override Drawable CreateContent() => text = CreateTextContainer();
+            protected virtual TextContainer CreateTextContainer() => new TextContainer();
+
+            protected class TextContainer : Container, IHasText
+            {
+                public string Text
+                {
+                    get { return NormalText.Text; }
```
Comment (AlFasGD) [inline GDEdit/GDE.App/Main/UI/Toolbar/ToolbarMenu.cs]: `=>`

File: GDEdit/GDEdit/Application/Database.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace GDEdit.Application
 {
     public static class Database
     {
-        public static readonly string GDLocalData = $@"{Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)}\GeometryDash";
+        public static readonly string GDLocalData = $@"{Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData)}\gdEditDevTest";
```
Comment (AlFasGD) [inline GDEdit/GDEdit/Application/Database.cs]: Revert this

File: GDEdit/GDE.Tests/Visual/TestCaseObject.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using osu.Framework.Graphics;
+using osu.Framework.Testing;
+using osu.Framework.Graphics.Sprites;
+using System.ComponentModel;
+using GDE.App.Main.Screens.Edit;
+using static GDEdit.Utilities.Functions.GeometryDash.Gamesave;
+using GDEdit.Application;
+using GDE.App.Main.Objects;
```
Comment (AlFasGD) [inline GDEdit/GDE.Tests/Visual/TestCaseObject.cs]: Sort usings (using static always goes after all the other usings)

File: GDEdit/GDE.Tests/Visual/TestCaseObject.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        public TestCaseObject()
+        {
+            Children = new Drawable[]
+            {
+                Obj = new ObjectBase
+                {
+                    Origin = Anchor.Centre,
+                    Anchor = Anchor.Centre,
+                    ID = 1,
+                    Size = new osuTK.Vector2(100)
```
Comment (AlFasGD) [inline GDEdit/GDE.Tests/Visual/TestCaseObject.cs]: `using osuTK;`?

File: GDEdit/GDE.App/Main/Tools/RavenLogger.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+using System.Threading.Tasks;
+using osu.Framework.Logging;
+using SharpRaven;
+using SharpRaven.Data;
+
+namespace GDE.App.Main.Tools
+{
+    public class RavenLogger : IDisposable
+    {
+        private readonly RavenClient raven = new RavenClient("https://5668996964374ccd8a5e288cc08567c5@sentry.io/1362706");
```
[the remark this answers is not in the document]
Comment (AltenGD) [reply GDEdit/GDE.App/Main/Tools/RavenLogger.cs]: its for sending the server to a secure site where we can fix the bug

File: GDEdit/GDE.App/Main/Objects/ObjectBase.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        private void load(TextureStore texStore)
+        {
+            switch(ID)
+            {
+                case 1:
+                    Console.WriteLine(Object.Texture = texStore.Get("Objects/square_01_001.png"));
+                    break;
+                default:
+                    throw new Exception("Rest hasnt been done yet");
+            }
```
[the remark this answers is not in the document]
Comment (AltenGD) [reply GDEdit/GDE.App/Main/Objects/ObjectBase.cs]: or can we just have the id as the name?

[the remark this answers is not in the document]
Comment (AltenGD) [reply GDEdit/GDE.App/Main/Objects/ObjectBase.cs]: ill try to do this in a later commit though

File: GDEdit/GDE.App/Main/Objects/ObjectBase.cs [hunk 7]
```diff
@@ -25,24 +25,52 @@ public class ObjectBase : Container
         ///<summary>Position Y of the object</summary>
         public double Y;
```
Comment (AltenGD) [inline GDEdit/GDE.App/Main/Objects/ObjectBase.cs]: To be continued
<=========

**Comment by AltenGD** [comment]
nice

**Pull request merged**